pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -579,6 +579,137 @@ def to_abstract_repr(self) -> str:
         validate_abstract_repr(abstr_dev_str, "device")
         return abstr_dev_str
 
+    def print_specs(self) -> None:
+        """Prints the device specifications."""
+        title = f"{self.name} Specifications"
+        header = ["-" * len(title), title, "-" * len(title)]
+        print("\n".join(header))
+        print(self._specs())
+
+    @property
+    def specs(self) -> str:
+        """Text summarizing the specifications of the device."""
+        return self._specs(for_docs=False)
+
+    def _specs(self, for_docs: bool = False) -> str:
+        def yes_no_fn(cond: bool) -> str:
+            return "Yes" if cond else "No"
+
+        lines = [
+            "\nRegister parameters:",
+            f" - Dimensions: {self.dimensions}D",
+            f" - Rydberg level: {self.rydberg_level}",
+            f" - Maximum number of atoms: {self.max_atom_num}",
+            f" - Maximum distance from origin: {self.max_radial_distance} μm",
+            (
+                " - Minimum distance between neighbouring atoms: "
+                f"{self.min_atom_distance} μm"
+            ),
+            f" - Maximum layout filling fraction: {self.max_layout_filling}",
+            f" - SLM Mask: {yes_no_fn(self.supports_slm_mask)}",
+        ]
+
+        if self.max_sequence_duration is not None:
+            lines.append(
+                " - Maximum sequence duration: "
+                f"{self.max_sequence_duration} ns"
+            )
+
+        device_lines = [
+            "\nDevice parameters:",
+        ]
+
+        if self.max_runs is not None:
+            device_lines.append(f" - Maximum number of runs: {self.max_runs}")
+        device_lines += [
+            f" - Channels can be reused: {yes_no_fn(self.reusable_channels)}",
+            f" - Supported bases: {', '.join(self.supported_bases)}",
+            f" - Supported states: {', '.join(self.supported_states)}",
+        ]
+
+        if self.interaction_coeff is not None:
+            device_lines.append(
+                f" - Ising interaction coefficient: {self.interaction_coeff}"
+            )
+
+        if self.interaction_coeff_xy is not None:
+            device_lines.append(
+                f" - XY interaction coefficient: {self.interaction_coeff_xy}"
+            )
+
+        if self.default_noise_model is not None:
+            device_lines.append(
+                f" - Default noise model: {self.default_noise_model}"
+            )
+
+        layout_lines = [
+            "\nLayout parameters:",
+            f" - Requires layout: {yes_no_fn(self.requires_layout)}",
+        ]
+
+        if hasattr(self, "accepts_new_layouts"):
+            layout_lines.append(
+                f" - Accepts new layout: {yes_no_fn(self.accepts_new_layouts)}"
+            )
```

Comment:
How would you feel about separating the different lines in private methods (ie `_reg_lines()`, `_device_lines()`, ...) and then have `_specs()` combine them? I think it would potentially improve readability and it would allow us to override `Device._layout_lines()` to add this extra parameter that is not part of `BaseDevice`.

Reply:
I can see you are not too keen on doing this, no worries :)

Reply:
Let me implement it, after your comment on using a string vs a list, I see more the advantage of doing it now, such that the code can be simplified further if we don't have to check if an object has an attribute.

Reply:
What do you suggest for the `yes_no_fn` and `check_none` helpers? Should I redefine them in each method? Or just go back to the previous form, where the test was done explicitly in each case?

Reply:
I'd prefer reusing them over redefining them. It's not the end of the world to have them as private functions at the end of the file. Between reusing and just doing things explicitly, I'll leave that up to you